ports/lua/CMakeLists.txt

```
@@ -1,117 +1,256 @@
-    ADD_EXECUTABLE ( luai ${CMAKE_SOURCE_DIR}/src/lua.c )  # interpreter
-
-    TARGET_INCLUDE_DIRECTORIES(luac PRIVATE ${CMAKE_CURRENT_LIST_DIR}/src)
-    TARGET_LINK_LIBRARIES ( luai PRIVATE lua )
-    SET_TARGET_PROPERTIES ( luai PROPERTIES OUTPUT_NAME lua PDB_NAME luai )
-    IF (UNIX)
-        IF (CMAKE_SYSTEM_NAME STREQUAL FreeBSD)
-            SET (_LIB_READLINE_NAME edit)
-        ELSE ()
-            SET (_LIB_READLINE_NAME readline)
-        ENDIF ()
-        FIND_LIBRARY (LIB_READLINE NAMES ${_LIB_READLINE_NAME})
-        IF (LIB_READLINE)
-            TARGET_COMPILE_DEFINITIONS (luai PUBLIC -DLUA_USE_READLINE)
-            TARGET_LINK_LIBRARIES(luai PRIVATE ${LIB_READLINE})
-            IF (_LIB_READLINE_NAME STREQUAL edit)
-                TARGET_INCLUDE_DIRECTORIES (luai PUBLIC /usr/include/edit)
-            ENDIF ()
-        ENDIF ()
-        
-        IF (LIB_MATH)
-            TARGET_LINK_LIBRARIES (luac m)
-        ENDIF ()
-    ENDIF ()
-    INSTALL ( TARGETS luai luac RUNTIME DESTINATION tools/lua )
-ENDIF ()
-
-IF (NOT SKIP_INSTALL_HEADERS)
-    INSTALL(
-        FILES
-            src/lualib.h
-            src/lua.h
-            src/luaconf.h
-            src/lauxlib.h
-        DESTINATION include
+    add_executable(
+        lua-compiler
+        src/luac.c
+        ${SRC_LIBLUA}
     )
-    # If using C++, don't install extern "C" wrapper.
-    IF (NOT COMPILE_AS_CPP)
-        INSTALL(FILES src/lua.hpp DESTINATION include)
-    ENDIF ()
-ENDIF ()
+    add_executable(unofficial::lua::lua-compiler ALIAS lua-compiler)
+
+    set_target_properties(
+        lua-compiler
+        PROPERTIES
+        OUTPUT_NAME luac
+    )
+
+    if (LIB_MATH)
+        target_link_libraries(lua-compiler m)
+    endif()
+
+    # interpreter
+    add_executable(
+        lua-interpreter
+        src/lua.c
+    )
+    add_executable(unofficial::lua::lua-interpreter ALIAS lua-interpreter)
+
+    target_link_libraries(
+        lua-interpreter
+        lua
+    )
+
+    set_target_properties(
+        lua-interpreter
+        PROPERTIES
+        OUTPUT_NAME lua
+        PDB_NAME luai
+    )
+
+    if (UNIX)
+        if (CMAKE_SYSTEM_NAME STREQUAL FreeBSD)
+            set(_LIB_READLINE_NAME edit)
+        else ()
+            set(_LIB_READLINE_NAME readline)
+        endif()
+        find_library(LIB_READLINE NAMES ${_LIB_READLINE_NAME})
+        if (LIB_READLINE)
+            target_compile_definitions(lua-interpreter PUBLIC LUA_USE_READLINE)
+            target_link_libraries(lua-interpreter ${LIB_READLINE})
+            if (_LIB_READLINE_NAME STREQUAL edit)
+                target_include_directories(lua-interpreter PUBLIC /usr/include/edit)
+            endif()
+        endif()
+    endif()
+
+    install(
+        TARGETS lua-interpreter lua-compiler
+        EXPORT unofficial-lua-targets
+        RUNTIME DESTINATION ${CMAKE_INSTALL_PREFIX}/tools/lua
```

Generally, do not use absolute destination paths with `install`:
```suggestion
        RUNTIME DESTINATION tools/lua
```
(It breaks `CMAKE_STAGING_PREFIX`. Not used in vcpkg.)

---

Okay, good to know.